cardano-api/src/Cardano/Api/TxMetadata.hs

```diff
@@ -100,15 +101,19 @@ instance HasTypeProxy TxMetadata where
 instance SerialiseAsCBOR TxMetadata where
     serialiseToCBOR =
           CBOR.serialize'
-        . Shelley.Metadata
+        . (Shelley.Metadata @())
+        -- The Shelley (Metadata era) is always polymorphic in era,
+        -- so we pick the unit type.
         . toShelleyMetadata
         . (\(TxMetadata m) -> m)
 
     deserialiseFromCBOR AsTxMetadata bs =
           TxMetadata
         . fromShelleyMetadata
         . (\(Shelley.Metadata m) -> m)
-      <$> CBOR.decodeAnnotator "TxMetadata" fromCBOR (LBS.fromStrict bs)
+      <$> CBOR.decodeAnnotator @(Shelley.Metadata ()) "TxMetadata" fromCBOR (LBS.fromStrict bs)
+        -- The Shelley (Metadata era) is always polymorphic in era,
+        -- so we pick the unit type.
```

Comment:
What are the implications of the ledger's metadate type being paramaterised by era? Is it actually going to vary in representation by era?

Reply:
It does not vary at all:

https://github.com/input-output-hk/cardano-ledger-specs/blob/06a46837fbde7c7148317ff359defa2b31da442b/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Metadata.hs#L72-L75

The era parameter was added just for consistency with the other eras, since `MetaData` is the Shelley era's notion of `AuxiliaryData`. Consistency was needed in order to make Shelley a `WellFormed` era: 
https://github.com/input-output-hk/cardano-ledger-specs/blob/06a46837fbde7c7148317ff359defa2b31da442b/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Metadata.hs#L72-L75